modules/extension/complex/src/test/java/org/geotools/filter/attribute/extractor/FilterAttributeExtractorTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2022, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.attribute.extractor;
+
+import static org.geotools.feature.FakeTypes.ANYSIMPLETYPE_TYPE;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.geotools.feature.NameImpl;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.feature.type.AttributeDescriptorImpl;
+import org.geotools.feature.type.AttributeTypeImpl;
+import org.geotools.feature.type.Types;
+import org.geotools.filter.AttributeExpressionImpl;
+import org.geotools.filter.FilterAttributeExtractor;
+import org.junit.Assert;
+import org.junit.Test;
+import org.opengis.feature.Feature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.feature.type.AttributeType;
+
+public class FilterAttributeExtractorTest {
+
+    @Test
+    public void testNonExistingAttributeEvaluatesToAttributeDescriptor() throws Exception {
+        List<AttributeDescriptor> schema = new ArrayList<>();
+        AttributeType doubleType =
+                new AttributeTypeImpl(
+                        new NameImpl("http://www.w3.org/2001/XMLSchema", "double"),
+                        Double.class,
+                        false,
+                        false,
+                        Collections.emptyList(),
+                        ANYSIMPLETYPE_TYPE,
+                        null);
+        schema.add(
+                new AttributeDescriptorImpl(
+                        doubleType, Types.typeName("pointOne"), 0, 1, false, null));
+        schema.add(
+                new AttributeDescriptorImpl(
+                        doubleType, Types.typeName("pointTwo"), 0, 1, false, null));
+        SimpleFeatureTypeBuilder sftb = new SimpleFeatureTypeBuilder();
```

Comment:
SimpleFeatureTypeBuilder has "add" methods to quickly create new attributes. Do you really need to create AttributeTypeImpl instances by hand? What for?